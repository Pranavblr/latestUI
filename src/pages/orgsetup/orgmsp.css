@media (min-width: 1024px) {
    
}

    
@media (max-width: 650px) {
    
}
    
@media screen and (max-width: 480px) {
   
}

.orgtitle {
    font-size: 30px;
    font-family: HoneywellSans-Medium;
    color: #303030;
    letter-spacing: 0;
    line-height: 98px;
    margin-left: 20px;
}

.orgsubtitle {
    font-family: HoneywellSans-Medium;
    font-size: 20px;
    color: #303030;
    letter-spacing: 0;
    margin-left: 5px;
}

#orglist > * {
    margin-left: 20px;
    margin-bottom: 20px;
}