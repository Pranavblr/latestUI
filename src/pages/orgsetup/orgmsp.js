import React, { Component } from "react";
import { connect } from "react-redux";
import { Button } from "@scuf/common";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import "./orgmsp.css";
import Dropdown from "react-dropdown";
import "react-dropdown/style.css";

export class OrgMspView extends Component {

    constructor(props) {
        super(props);
        this.state = { showAddOrg: false };
        this.handleShowAddOrg = this.handleShowAddOrg.bind(this);
    }

    handleShowAddOrg(){
        this.setState({showAddOrg: !this.state.showAddOrg})
    }
    render(){
        let {nwdata} = this.props
        console.log('props ',nwdata)
        let options = {}
        if (nwdata.length > 0 ) options = nwdata[0]
        const defaultOption = options[0]  
        
        return (
            <Container fluid={true}>
                
                <Row>
                    <div class='orgtitle'>Organizations</div>
                </Row>

                <Row id='orglist'>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org1</p>
                    </div>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org2</p>
                    </div>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org3</p>
                    </div>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org4</p>
                    </div>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org5</p>
                    </div>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org6</p>
                    </div>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org7</p>
                    </div>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org8</p>
                    </div>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org9</p>
                    </div>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org10</p>
                    </div>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org11</p>
                    </div>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org12</p>
                    </div>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org13</p>
                    </div>
                    <div>
                    <img src={process.env.PUBLIC_URL + '/images/org.png'}/>
                    <p align='center'>Org14</p>
                    </div>
                    
                </Row>

                <Row style={{"background-color": "#F7F7F7"}}>
                    <div style={{"background-color": "#1792E5", "margin-left": "15px"}}>
                        <Button class='addorgbtn' type="primary" iconRoot="building" icon="add" content="Add Org" iconPosition="left" onClick= { this.handleShowAddOrg }></Button>
                    </div>
                </Row>

                {this.state.showAddOrg && (
                    <div>
                    <Row>
                        <div class='orgtitle'>Add Organization</div>
                    </Row>
                    
                    <Row class='justify-content-start'>
                        <Col xs={12} md={3}>
                            <label class='orgsubtitle' for="name">Org Name</label>
                        </Col>
                        <Col xs={12} md={9}>
                            <input class='form-control' type="text" name="orgname" id="orgname" value=""/>
                        </Col>
                    </Row>
                    <Row>
                        <Col xs={12} md={3}>
                            <label class='orgsubtitle' for="msp">Org MSP ID</label>
                        </Col>
                        <Col xs={12} md={9}>
                            <Dropdown class='orgmsplist' options={options} onChange={this._onSelect} value={defaultOption} placeholder="Select an option" />
                        </Col>
                    </Row>
                    <Row>
                        <Col xs={12} md={3}>
                            <label class='orgsubtitle' for="msp">Org Admin ID</label>
                        </Col>
                        <Col xs={12} md={9}>
                        <Dropdown class='orgadminlist' options={options} onChange={this._onSelect} value={defaultOption} placeholder="Select an option" />
                        </Col>
                    </Row>
                    <Row>
                        <Col xs={12} md={3}>
                            <label class='orgsubtitle' for="msp">Org Admin Secret</label>
                        </Col>
                        <Col xs={12} md={9}>
                            <input class='form-control' type="text" name="orgadminsecret" id="orgadminsecret" value=""/>
                        </Col>
                    </Row>
                    <Row style= {{"justify-content" : "center"}}>
                    <Row>
                        <Col xs={12} md={6}>
                        <Button type="primary" content="CANCEL" onClick={ this.handleShowAddOrg }></Button>
                        </Col>
                        <Col xs={12} md={6}>
                        <Button type="primary" content="SUBMIT"></Button>
                        </Col>
                    </Row>
                    </Row>
                    </div>
                ) }   
            
            </Container>
        );
    }
}

const mapStateToProps = state => {
  console.log("orgmspview mapStateToProps called ", state);

  if (state.network.data == undefined) {
    return { nwdata: {} };
  }
    var orglist1 = state.network.data.network[0].channels[0].orgs.map( (item, index) => item.participantid)
    var orglist2 = state.network.data.network[1].channels[0].orgs.map( (item, index) => item.participantid)
    console.log('list1 ', orglist1, 'list2 ',orglist2)
    return { nwdata: [orglist1, orglist2] }
};

export default connect(mapStateToProps)(OrgMspView);
