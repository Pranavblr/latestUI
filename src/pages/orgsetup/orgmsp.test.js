import renderer from 'react-test-renderer';
import { OrgMspView } from './orgmsp'
import React from 'react';
import { shallow, mount } from 'enzyme';

const testdata = {
    "nwdata" : [["org1","org2","org3"], ["org4","org5","org6"]]
}

describe('should render orgmsp component properly', () => {


    it('org list should be displayed in combobox items', () => {

        const wrapper = mount(<OrgMspView nwdata={testdata}/>);
        const addorgbtn = wrapper.find('Button.addorgbtn')
        wrapper.setState({showAddOrg: true});
        addorgbtn.simulate('click')
        const dropdown = wrapper.find('Dropdown.orgadminlist')
        console.log('wrapper ', wrapper)
        expect(dropdown.props.options).toEqual(["org1","org2","org3"]);

    });

    it('org list should be displayed in combox items', () => {
        const component = renderer.create(
            <OrgMspView nwdata={testdata} />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });


})