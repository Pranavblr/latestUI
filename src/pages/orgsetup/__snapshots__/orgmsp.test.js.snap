// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render orgmsp component properly org list should be displayed in combox items 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      class="orgtitle"
    >
      Organizations
    </div>
  </div>
  <div
    className="row"
    id="orglist"
  >
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org1
      </p>
    </div>
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org2
      </p>
    </div>
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org3
      </p>
    </div>
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org4
      </p>
    </div>
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org5
      </p>
    </div>
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org6
      </p>
    </div>
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org7
      </p>
    </div>
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org8
      </p>
    </div>
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org9
      </p>
    </div>
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org10
      </p>
    </div>
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org11
      </p>
    </div>
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org12
      </p>
    </div>
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org13
      </p>
    </div>
    <div>
      <img
        src="/images/org.png"
      />
      <p
        align="center"
      >
        Org14
      </p>
    </div>
  </div>
  <div
    className="row"
    style={
      Object {
        "background-color": "#F7F7F7",
      }
    }
  >
    <div
      style={
        Object {
          "background-color": "#1792E5",
          "margin-left": "15px",
        }
      }
    >
      <button
        class="addorgbtn"
        className="ui medium button primary"
        onClick={[Function]}
        style={
          Object {
            "minWidth": "auto",
          }
        }
        type="button"
      >
        <i
          aria-label=""
          className="h-icon building add left"
          name="add"
          style={Object {}}
          title=""
        />
        Add Org
      </button>
    </div>
  </div>
</div>
`;
